src/commands/install-deps.yml

```diff
@@ -45,22 +38,35 @@ steps:
           APP_BUNDLER_VERSION=<<parameters.bundler-version>>
         fi
 
-        if ! echo $(bundle version)  | grep -q $APP_BUNDLER_VERSION; then
+        if ! echo $(bundle version) | grep -q $APP_BUNDLER_VERSION; then
           echo "Installing bundler $APP_BUNDLER_VERSION"
           gem install bundler:$APP_BUNDLER_VERSION
         else
           echo "bundler $APP_BUNDLER_VERSION is already installed."
         fi
 
+        test -f Gemfile.lock || bundle lock
+
+  - when:
+      condition: <<parameters.with-cache>>
+      steps:
+        - restore_cache:
+            keys:
+              - << parameters.key >>-{{ .Branch }}-{{ checksum "./Gemfile.lock"  }}
+              - << parameters.key >>-{{ .Branch }}
+              - << parameters.key >>-
+  - run:
+      name: Bundle Install <<^parameters.with-cache>>(No Cache)<</parameters.with-cache>>
+      command: |
         if "<< parameters.path >>" == "./vendor/bundle"; then
-          bundle check --path <<parameters.path>> || bundle install --deployment
+          bundle check --path <<parameters.path>> || (bundle install --deployment && bundle clean)
```

Comment:
Does `bundle clean` have any cost? Otherwise, this seems like a great way to keep the cache tidy!